### PR TITLE
Integrate libFirm into our compiler

### PR DESCRIPTION
**Pull request by g4rgoil**
Open questions:

- [ ] Should we allow for libFirm build variants other than `debug`?
- [x] ~Should we somehow integrate building of libFirm into Maven?~

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -32,17 +35,17 @@ jobs:
         with:
           python-version: '3.6.9' # Version range or exact version of a Python version to use, using SemVer's version range syntax
           architecture: 'x64' # optional x64 or x86. Defaults to x64 if not specified
+      # - name: Init Submodules
+      #   run: |
+      #     pwd
+      #     git submodule init
+      #     git submodule update
```
Comment (N-Maas) [inline .github/workflows/maven.yml:42]: I assume we can remove this?

**Comment by N-Maas** [comment]
> Open questions:
> 
>     *   Should we allow for libFirm build variants other than `debug`?
> 
>     *  Should we somehow integrate building of libFirm into Maven?

Regarding the second question, I don't think this is necessary.

**Review by N-Maas — APPROVED** [review]
I think we can merge it in the current state and think about other build variants later.

**Comment by g4rgoil** [comment]
> I think we can merge it in the current state and think about other build variants later.

That's fine by me, the CI issues should be sorted out.

**Comment by 0xNB** [comment]
Is there a list of variants somehwere? The `README` is pretty sparse

**Comment by 0xNB** [comment]
Is there a list of variants somehwere? The `README` is pretty sparse

**Comment by 0xNB** [comment]
Is there a list of variants somehwere? The `README` is pretty sparse

**Comment by 0xNB** [comment]
Is there a list of variants somehwere? The `README` is pretty sparse

**Comment by 0xNB** [comment]
Is there a list of variants somehwere? The `README` is pretty sparse

**Comment by g4rgoil** [comment]
> Is there a list of variants somehwere? The `README` is pretty sparse

There is no list as such, you kind of need to dig into their `Makefile`. The relevant variant for us would probably be `optimize` which runs `gcc` with `-O3`.

**Pull request merged**